src/coreclr/tests/src/JIT/HardwareIntrinsics/Arm/AdvSimd/DuplicateSelectedScalarToVector128.V128.Int16.4.cs

```diff
@@ -0,0 +1,506 @@
+
+                return testStruct;
+            }
+
+            public void RunStructFldScenario(ImmUnaryOpTest__DuplicateSelectedScalarToVector128_V128_Int16_4 testClass)
+            {
+                var result = AdvSimd.DuplicateSelectedScalarToVector128(_fld, 4);
+
+                Unsafe.Write(testClass._dataTable.outArrayPtr, result);
+                testClass.ValidateResult(_fld, testClass._dataTable.outArrayPtr);
+            }
+
+            public void RunStructFldScenario_Load(ImmUnaryOpTest__DuplicateSelectedScalarToVector128_V128_Int16_4 testClass)
+            {
+                fixed (Vector128<Int16>* pFld = &_fld)
+                {
+                    var result = AdvSimd.DuplicateSelectedScalarToVector128(
+                        AdvSimd.LoadVector128((Int16*)(pFld)),
+                        4
+                    );
+
+                    Unsafe.Write(testClass._dataTable.outArrayPtr, result);
+                    testClass.ValidateResult(_fld, testClass._dataTable.outArrayPtr);
+                }
+            }
+        }
+
+        private static readonly int LargestVectorSize = 16;
+
+        private static readonly int Op1ElementCount = Unsafe.SizeOf<Vector128<Int16>>() / sizeof(Int16);
+        private static readonly int RetElementCount = Unsafe.SizeOf<Vector128<Int16>>() / sizeof(Int16);
+
+        private static Int16[] _data = new Int16[Op1ElementCount];
+
+        private static Vector128<Int16> _clsVar;
+
+        private Vector128<Int16> _fld;
+
+        private DataTable _dataTable;
+
+        static ImmUnaryOpTest__DuplicateSelectedScalarToVector128_V128_Int16_4()
+        {
+            for (var i = 0; i < Op1ElementCount; i++) { _data[i] = TestLibrary.Generator.GetInt16(); }
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<Vector128<Int16>, byte>(ref _clsVar), ref Unsafe.As<Int16, byte>(ref _data[0]), (uint)Unsafe.SizeOf<Vector128<Int16>>());
+        }
+
+        public ImmUnaryOpTest__DuplicateSelectedScalarToVector128_V128_Int16_4()
+        {
+            Succeeded = true;
+
+            for (var i = 0; i < Op1ElementCount; i++) { _data[i] = TestLibrary.Generator.GetInt16(); }
+            Unsafe.CopyBlockUnaligned(ref Unsafe.As<Vector128<Int16>, byte>(ref _fld), ref Unsafe.As<Int16, byte>(ref _data[0]), (uint)Unsafe.SizeOf<Vector128<Int16>>());
+
+            for (var i = 0; i < Op1ElementCount; i++) { _data[i] = TestLibrary.Generator.GetInt16(); }
+            _dataTable = new DataTable(_data, new Int16[RetElementCount], LargestVectorSize);
+        }
+
+        public bool IsSupported => AdvSimd.IsSupported;
+
+        public bool Succeeded { get; set; }
+
+        public void RunBasicScenario_UnsafeRead()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_UnsafeRead));
+
+            var result = AdvSimd.DuplicateSelectedScalarToVector128(
+                Unsafe.Read<Vector128<Int16>>(_dataTable.inArrayPtr),
+                4
```

Comment:
>4 [](start = 16, length = 1)

sorry, i might have missed it, but I am not seeing a test where the index element is a non-constant.

Reply:
Its covered by the reflection scenario

Reply:
I see. Is it in some other file? This file still takes `4` in `RunReflectionScenario_UnsafeRead` and `RunReflectionScenario_Load`

Reply:
ah, the parameters during invocation will be non-constant I assume.

Reply:
Yes, reflection requires boxing everything and performing an indirect invocation, so it being a constant is completely lost.